src/west/commands/help.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''West common commands'''
+
+from west.commands import WestCommand
+from west.util import WestReload
+
+class Help(WestCommand):
+    def __init__(self):
+        super().__init__(
+            'help',
+            'print west command help text',
+            'Print help text on any command.'
+        )
+
+    def do_add_parser(self, parser_adder):
+        return self.add_parser_helper(
+            parser_adder,
+            self.add_arg('command', metavar='cmd', nargs='?',
+                         help='Command to get help on')
+        )
+
+    def do_run(self, args, user_args):
+        a = [args.command, '-h']
+        wr = WestReload(argv=[x for x in a if x is not None])
+        raise wr
```

Comment:
Just doing some more thinking here.
Are there any reasons why we explicitly need to relaunch `west` ?
It should be possible to just call into `west.main` as we do with bootstrapper and wrapped commands.

The reason why `west update` has to relaunch, is because the `west` source code have changed when the SHAs differs, and thus we need to run rest of the upgrade process using the new `west` code.

Reply:
Not relaunching `west` could also make it easier to fix this https://github.com/zephyrproject-rtos/west/pull/204#discussion_r258152013

Reply:
This seems to work the same, and avoids the overhead of an exception and relaunching:
```
        import west.main
        west.main.main(argv=[x for x in a if x is not None])
```
instead of:
```
       wr = WestReload(argv=[x for x in a if x is not None])
       raise wr
```
Disclaimer, not fully tested at this point.

Reply:
Note: `west help init` will need special handling with suggested approach, as `west init -h` is solely handled in the bootstrapper today.

Reply:
I am very reluctant to do that because there's a whole bunch of things that are initialized and running inside the same Python VM instance makes me uncomfortable. Also it is very ugly to call back into what just called you IMHO?
@mbolivar thoughts?

Reply:
@carlescufi in general I agree with you, but with the `west help` we know exactly where we are.
And seeing the `west help <cmd>` as a wrapper for `west <cmd> -h` I think that is nicer in this particular case.

I think it might be possible to fetch directly into `<command>.arg_parser` to get the help text we need.
That could make it an even nicer solution, than calling `west.main.main`
@mbolivar You have bit more experience with arg parser, would should approach seem feasible ?

My main point is that we don't need to relaunch.